src/main/java/org/folio/rest/impl/InventoryHelper.java

```diff
@@ -626,4 +644,28 @@ private JsonObject getFirstObjectFromResponse(JsonObject response, String proper
                    .map(item -> (JsonObject) item)
                    .orElseThrow(() -> new CompletionException(new InventoryException(String.format("No records of '%s' can be found", propertyName))));
   }
+
+  /**
+   * Returns JsonObject with id of ContributorNameType retrieved from inventory-storage.
+   *
+   * @param contributorNameTypeName ContributorNameType
+   * @return JsonObject ContributorNameTypeId
+   */
+  private CompletableFuture<JsonObject> getContributorNameTypeId(ContributorNameTypeName contributorNameTypeName) {
+    String name = contributorNameTypeName.getContributorNameTypeName();
+    String endpoint = String.format("/contributor-name-types?query=name==%s", encodeQuery(name, logger));
+    if(ctx.get(name) == null) {
```

Comment:
Please add the same logic to `getInstanceType` and `getStatus` just to be consistent

Reply:
Hi @piotr-kalashuk,
this case I can suggest generalizing of retrieving and caching logic for `getInstanceType`, `getStatus` and `getContributorNameTypeId`.